src/Moryx.AbstractionLayer.Products.Endpoints/ProductManagementController.cs

```diff
@@ -240,10 +241,18 @@ public ActionResult<ProductModel> Duplicate(long id, [FromBody] string newIdenti
             if (identityArray.Length != 2)
                 return BadRequest($"Identity has wrong format. Must be identifier-revision");
             var identity = new ProductIdentity(identityArray[0], Convert.ToInt16(identityArray[1]));
-            var newProductType = _productManagement.Duplicate(template, identity);
-            if (newProductType == null)
-                return BadRequest($"Error while duplicating");
-            return _productConverter.ConvertProduct(newProductType, false);
+
+            try
+            {
+                var newProductType = _productManagement.Duplicate(template, identity);
+                if (newProductType == null)
```

Comment:
You would want to limit the try catch block only to the relevant code section, which I think is only the `_productManagement.Duplicate(template, identity);` block. Otherwise, great solution 👍